crates/subspace-farmer/src/plot/worker.rs

```diff
@@ -167,13 +167,24 @@ impl<T: PlotFile> PlotWorker<T> {
                             };
                             let _ = result_sender.send(result);
                         }
-                        Request::ReadPieceIndexes {
+                        Request::ReadSequentialPieces {
```

Comment:
I understand how this is beneficial, but if it takes a few seconds, it is a huge problem because it will potentially prevent farmer from returning a solution while this read is happening. We need to make sure we enforce limits on this, potentially requesting less pieces than we would have requested otherwise to slice work into smaller chunks and allow solving to read pieces in case farmer happens to solve successfully.

Reply:
> Or we could divide one request for peer into several requests for plot worker.

I think this.